### ISSUE TITLE
Handling of Mosaics in MSv4

### ISSUE THREAD

**Issue by dpetry** [casangi/xradio#314]
As discussed with Jan-Willem, the MSv4 originally was meant to contain data for only one Field
such that multi-field MSs would need to be stored in several MSv4 inside one Processing Set.
Later the MSv4 was modified to permit more than one Field.
So, for **mosaics**, i.e. observations of the same target with the same array in one execution block
sequentially moving over a **set of Fields** (in the ASDM and MSv2, they all have the same Field Name
but not the same Field ID), the questions are:
1) Do the data belonging to the many fields (potentially hundreds) of the same mosaic all get stored 
in _one_ MSv4 or do they get spread over many MSv4s?
What should be the default behaviour?
2) How can a piece of software reading the data recognise that it is dealing with a Mosaic?
3) How can the reading software determine the extend of the Mosaic (total number of fields, spatial extent)?
    In other words, how is the metadata of the mosaic set in the first place and then made available?

**Comment by Jan-Willem** [comment]
- "in the ASDM and MSv2, they all have the same Field Name but not the same Field ID" The converter will append field_ids to the names so that the different fields are unique for example ['NGC4038 - Antennae North_1',..., 'NGC4038 - Antennae North_9']. Selection of all the 'NGC4038 - Antennae North' fields are still possible using sub_ps=ps.sel(field_name='NGC4038 - Antennae North',string_exact_match=False).

1. Partitioning on field is optional using the partition_schema parameter. The current default is to partition on fields which would include the fields in a mosaic. We will have to experiment with this to determine what the recommendation is for ALMA mosaics.
2. The software would see that for the OBSERVE TARGET#ON_SOURCE intent, there is more than one field_name. We can also add is_mosaic attribute to the field_and_source_xds (we already have this for ephemeris data).
3. The processing set has the methods get_combined_field_and_source_xds and plot_phase_centers that will give some of this information. Note that these functions are not complete, they were created to demonstrate the type of functionality we can add to the PS.

**Comment by FedeMPouzols** [comment]
I think this is clarified. The MSv4 Review Committee considered this issue/RID closed without further action.

Please re-open or open a related issue if there is anything else to discuss or fix.
